include/alpaka/core/RemoveRestrict.hpp

```diff
@@ -0,0 +1,30 @@
+/* Copyright 2021 Rene Widera
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+namespace alpaka
+{
+    //! Removes __restrict__ from a type
+    template<typename T>
+    struct remove_restrict
+    {
+        using type = T;
+    };
+
+    template<typename T>
+    struct remove_restrict<T __restrict__>
+    {
+        using type = T;
+    };
```

Comment:
I am pretty sure you will need to guard this with `#ifdef`s for all the various spellings and supports on various compilers. AFAIK MSVC calls it `__restrict`. I also wonder if all compilers implement the same semantic and make restrict part of the type system. Let's see what the CI says.

Reply:
So then alpaka has to define some abstraction macro for `restrict`?

Reply:
Not necessarily. With a look at the recent CI outcome, an `#ifdef` for MSVC might actually be enough here. On the other hand, a more portable abstraction for `restrict` could have a place in alpaka.